docs/language/informal/reserve-await-and-yield.md

```diff
@@ -0,0 +1,244 @@
+    code into a flat sequence of tokens and use only local information to decide
+    how to color each one. That means they don't have the nested tree structure
+    neeeded to tell if a word like `on` is appearing in a catch clause and acts
+    like a keyword or is used as an identifier elsewhere.
+
+    The parsing framework, Grammar-Kit, used by the IntelliJ plug-in has
+    problems handling them.
+
+*   **It's harder to produce good error messages around them.** When an
+    erroneous piece of code uses a contextual keyword, it may not be clear if
+    the author was trying to use it as a keyword, or an identifier. Usually, a
+    smart tool can guess correctly, but it's still a guess. The error message
+    usually has to be worded to handle both cases, which tends to make it
+    confusing.
+
+## `await` and `yield`
+
+The last two problems apply in particular to `await` and `yield`. Most
+contextual keywords can be disambiguated using very local context. `hide` is
+only a keyword when right inside an import directive. `of` is only a keyword if
+it immediately follows `part` at the top level.
+
+But `await` is a keyword if the surrounding method is declared `async`. That
+modifier may be very far away from the use of `await`. `yield` is similar for
+`sync` methods.
+
+What makes this worse is that it is a very common error to forget the `async`
+modifier when intending to write an asynchronous method. So the error message
+around misusing `await` is both hard to write well and actually occurs very
+often in practice.
+
+The proposal is simple: Add `await` and `yield` to the list of reserved words.
+It is a syntax error to use either as an identifier, even outside of async or
+generator functions.
+
+**This is a breaking change.** However, `await` and `yield` are rarely used as
+identifiers, so the practical impact of the breakage is likely minimal. In a
+corpus of 49,553 Dart files, including the Dart repository, Flutter, and the
+latest versions of all packages on Pub, I found:
+
+* 47,457 uses of `await` as a keyword.
+* 14 uses of `await` as an identifier.
+* 878 uses of `yield` as a keyword.
+* 48 uses of `yield` as an identifier.
+* 13,418,913 uses of any identifier.
+
+In other words, 0.0001% of identifiers are `await` and 0.0003% are `yield`. A
+random use of `await` is 3389 times more likely to be a keyword than an
+identifier.
+
+Even so, we may wish to roll this out as part of the transition to Dart 2.0.
+
+## What about other contextual keywords?
+
+That still leaves a long list of contextual keywords. Should we reserve any of
+those while we're at it? The tools folks who requested we reserve `await` and
+`yield` also requested `async`. However:
+
+*   "async" is the name of a prominent Dart library, "dart:async". If we ever
+    get a [syntax for imports that doesn't require a quoted string][import],
+    that would cause problems if `async` was a reserved word.
```

Comment:
Probably not. We will not require library names to be identifiers, just character sequences delimited by context.
I.e., this is not a problem for us, but it may still be one for syntax highlighters.